### PR TITLE
Alliance setup

### PR DESCRIPTION
**Pull request by agdust**
Oof, that was not easy :D

Seems like all is working as it should, but there are few things that need to be mentioned.
1. In the Deck.svelte I had to make reference to current deck (state.cards), but we need a constant card info reference. So we won't need to duplicate data in Rust and JS
2. I've hid other Alliance components on the faction board for now, since it's need to be fully tested and I don't think I will be able to do it in nearest future

Looks like I'm starting to understand how lumber works (for this project at least), but I still have some questions, and will figure them out and ask soon 🙉

### CODE REVIEW

File: game/game/cards.lumber [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+:- pub(card/3).
+card(G, Id, Kind) :-
+  G =:= game { cards: Cards, discards: Discards, .. },
+  Card =:= card { id: Id, card: card(Kind), .. },
+  in(Card, Cards),
+  notin(Card, Discards).
+
+:- pub(cardSuit/3).
+cardSuit(game { cards: Cards, ..}, Id, Suit ) :-
+  in(card { id: Id, suit: !suit(Suit), .. }, Cards).
```
Comment (foxfriends) [inline game/game/cards.lumber:19]: Are these being used anywhere? I think not? I think they can be the same function anyway, which just checks the details of a card by ID, not worrying about who owns it or if its discarded. `isDiscarded/2` can be a separate one if needed.

```
card(G, Id, Kind, Suit)
```

File: game/setup/mod.lumber [hunk 2]
```diff
@@ -33,6 +36,15 @@ perform(Name, State, Action, NewState) :-
...
 
+// actions(Name, State, Action) :-
+//     turn(State, alliance),
+//     playerFaction(State, Name, alliance),
+//     alliance::actions(Name, State, Action).
+// perform(Name, State, Action, NewState) :-
+//     turn(State, alliance),
+//     playerFaction(State, Name, alliance),
+//     alliance::perform(Name, State, Action, NewState).
+
```
Comment (foxfriends) [inline game/setup/mod.lumber:47]: Don't need these, can just remove.

```suggestion
```

**Review by foxfriends — COMMENTED** [review]
Looks good! Mostly on me here for not figuring out how to do cards better yet, they're kinda weird as is. I will think about it some more...

File: game/setup/mod.lumber [hunk 3]
```diff
@@ -4,15 +4,18 @@
...
 :- pub(actions/3).
 :- pub(perform/4).
 
 nextSetup(G, S) :- (
-    turn(G, eyrie) ->> eyrie::init(G, S);
-    G =:= S
+    turn(G, eyrie) ->> eyrie::init(G, S); (
+        turn(G, alliance) ->> alliance::init(G, S);
+        G =:= S
+    )
```
[the remark this answers is not in the document]
Comment (agdust) [reply game/setup/mod.lumber:18]: Nope, just tried, looks like this syntax is not the same 🤔

**Review by foxfriends — COMMENTED** [review]
Thanks!

I will probably merge this this weekend when I have a chance to look into why the double `->>` is not working

**Pull request merged**